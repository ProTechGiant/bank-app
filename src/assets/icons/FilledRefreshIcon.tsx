import Svg, { Circle, ClipPath, Defs, G, Path, Rect } from "react-native-svg";

import type { IconProps } from "@/assets/icons";

export function FilledRefreshIcon({ width = 24, height = 24, color = "#FFF0DC" }: IconProps) {
  return (
    <Svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <Circle cx="12" cy="12" r="12" fill="black" />
      <G clip-path="url(#clip0_3931_21262)">
        <Path
          d="M12 7.5V5.8125L9.75 8.0625L12 10.3125V8.625C13.8619 8.625 15.375 10.1381 15.375 12C15.375 12.5681 15.2344 13.1081 14.9812 13.575L15.8025 14.3963C16.2413 13.7044 16.5 12.8831 16.5 12C16.5 9.51375 14.4863 7.5 12 7.5ZM12 15.375C10.1381 15.375 8.625 13.8619 8.625 12C8.625 11.4319 8.76562 10.8919 9.01875 10.425L8.1975 9.60375C7.75875 10.2956 7.5 11.1169 7.5 12C7.5 14.4863 9.51375 16.5 12 16.5V18.1875L14.25 15.9375L12 13.6875V15.375Z"
          fill={color}
        />
      </G>
      <Defs>
        <ClipPath id="clip0_3931_21262">
          <Rect width="13.5" height="13.5" fill="#ffffff" transform="translate(5.25 5.25)" />
        </ClipPath>
      </Defs>
    </Svg>
  );
}
