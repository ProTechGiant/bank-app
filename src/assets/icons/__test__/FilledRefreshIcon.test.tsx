import { create } from "react-test-renderer";

import { FilledRefreshIcon } from "../FilledRefreshIcon";

describe("icon/FilledRefreshIcon", () => {
  it("renders a FilledRefreshIcon", () => {
    const result = create(<FilledRefreshIcon />);

    expect(result).toMatchSnapshot();
  });
});
