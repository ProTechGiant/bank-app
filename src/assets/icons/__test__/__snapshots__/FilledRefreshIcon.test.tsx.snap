// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/FilledRefreshIcon renders a FilledRefreshIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGCircle
      cx="12"
      cy="12"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
      r="12"
    />
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGPath
        d="M12 7.5V5.8125L9.75 8.0625L12 10.3125V8.625C13.8619 8.625 15.375 10.1381 15.375 12C15.375 12.5681 15.2344 13.1081 14.9812 13.575L15.8025 14.3963C16.2413 13.7044 16.5 12.8831 16.5 12C16.5 9.51375 14.4863 7.5 12 7.5ZM12 15.375C10.1381 15.375 8.625 13.8619 8.625 12C8.625 11.4319 8.76562 10.8919 9.01875 10.425L8.1975 9.60375C7.75875 10.2956 7.5 11.1169 7.5 12C7.5 14.4863 9.51375 16.5 12 16.5V18.1875L14.25 15.9375L12 13.6875V15.375Z"
        fill={
          {
            "payload": 4294963420,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
    <RNSVGDefs>
      <RNSVGClipPath
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        name="clip0_3931_21262"
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          height="13.5"
          matrix={
            [
              1,
              0,
              0,
              1,
              5.25,
              5.25,
            ]
          }
          propList={
            [
              "fill",
            ]
          }
          width="13.5"
          x="0"
          y="0"
        />
      </RNSVGClipPath>
    </RNSVGDefs>
  </RNSVGGroup>
</RNSVGSvgView>
`;
